packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,631 @@
+  }
+
+  ['@test it should assert unbound cannot be called with multiple arguments']() {
+    let willThrow = () => {
+      this.render(`{{unbound foo bar}}`, {
+        foo: 'BORK',
+        bar: 'BLOOP'
+      });
+    };
+
+    expectAssertion(willThrow, /unbound helper cannot be called with multiple params or hash params/);
+  }
+
+  ['@test should render on attributes']() {
+    this.render(`<a href="{{unbound foo}}"></a>`, {
+      foo: 'BORK'
+    });
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => this.rerender());
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => set(this.context, 'foo', 'OOF'));
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => set(this.context, 'foo', 'BORK'));
+
+    this.assertHTML('<a href="BORK"></a>');
+  }
+
+  ['@htmlbars should property escape unsafe hrefs']() {
+    this.render(`<ul>{{#each people as |person|}}<li><a href="{{unbound person.url}}">{{person.name}}</a></li>{{/each}}</ul>`, {
+      people: emberA([
+        {
+          name: 'Bob',
+          url: 'javascript:bob-is-cool' // jshint ignore:line
+        },
+        {
+          name: 'James',
+          url: 'vbscript:james-is-cool' // jshint ignore:line
+        },
+        {
+          name: 'Richard',
+          url: 'javascript:richard-is-cool' // jshint ignore:line
+        }
+      ])
+    });
+
+    this.assertHTML(strip`
+      <ul>
+        <li>
+          <a href="unsafe:javascript:bob-is-cool">Bob</a>
+        </li>
+        <li>
+          <a href="unsafe:vbscript:james-is-cool">James</a>
+        </li>
+        <li>
+          <a href="unsafe:javascript:richard-is-cool">Richard</a>
+        </li>
+      </ul>
+    `);
+  }
+
+  ['@test it should render the current value of a property path on the context with container']() {
+    let owner = buildOwner();
+    this.render(`{{unbound foo}}`, {
+      [OWNER]: owner,
+      foo: 'bleep'
+    });
+
+    this.assertText('bleep');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bleep');
+
+    this.runTask(() => set(this.context, 'foo', 'bloop'));
+
+    this.assertText('bleep');
+
+    this.runTask(() => set(this.context, 'foo', 'bleep'));
+  }
+
+  ['@htmlbars helper form updates on parent re-render']() {
+    this.render(`{{unbound foo}}`, {
+      foo: 'BORK'
+    });
+
+    this.assertText('BORK');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('BORK');
+
+    this.runTask(() => set(this.context, 'foo', 'OOF'));
+
+    this.assertText('BORK');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('OOF');
+
+    this.runTask(() => set(this.context, 'foo', ''));
+
+    this.assertText('OOF');
+
+    this.runTask(() => set(this.context, 'foo', 'BORK'));
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('BORK');
+  }
+
+  // semantics here is not guaranteed
+  ['@test sexpr form does not update does unless explicitly told to']() {
```

Comment:
typo: "does not update does" (I *think* this is actually testing it doesn't update *no matter what*?)